fsspec/spec.py

```diff
@@ -878,12 +878,18 @@ def get(self, rpath, lpath, recursive=False, callback=_DEFAULT_CALLBACK, **kwarg
 
         Calls get_file for each source.
         """
-        from .implementations.local import make_path_posix
+        from .implementations.local import LocalFileSystem, make_path_posix
 
         if isinstance(lpath, str):
             lpath = make_path_posix(lpath)
         rpaths = self.expand_path(rpath, recursive=recursive)
-        lpaths = other_paths(rpaths, lpath)
+        isdir = isinstance(lpath, str) and LocalFileSystem().isdir(lpath)
+        lpaths = other_paths(
+            rpaths,
+            lpath,
+            exists=isdir and isinstance(rpath, str) and not rpath.endswith("/"),
```

Comment:
Is the path ever not a string here, do we test this?

Reply:
Yes and yes. The path can be a sequence of strings, e.g. `get(["file1", "file2"], "target_dir")`, and the multiple paths case is tested in the new `test_cp_two_files` functions in both `test_local.py` and `test_memory.py`.

Reply:
Got it, thanks. Good to be reminded of my design choices :)